stablehlo/reference/Index.h

```diff
@@ -34,10 +34,11 @@ class IndexSpaceIterator {
   /// \name Constructor
   IndexSpaceIterator(Sizes shape, std::optional<Index> index)
       : shape_(shape), index_(index) {
-    if (index && !index->inBounds(shape))
-      llvm::report_fatal_error(
-          "Incompatible index and shape found while creating "
-          "an IndexSpaceIterator");
+    if (index && !index->inBounds(shape)) index_ = std::nullopt;
+  }
+
+  IndexSpaceIterator(Sizes shape) : shape_(shape) {
```

Comment:
Not sure why we need this. 

The earlier constructor has the provision of a `std::nullopt` to index param, which would initialize `index_` the way this constructor is doing. To me it seems the constructor is introduce to avoid explicitly passing that `std::nullopt`.

Reply:
Thanks for pointing it out, I meant to remove this.